internal/gcsx/multi_range_downloader_wrapper.go

```diff
@@ -105,37 +116,98 @@ func (mrdWrapper *MultiRangeDownloaderWrapper) GetRefCount() int {
 // IncrementRefCount increments the refcount and cancel any running cleanup function.
 // This method should be called exactly once per user of this wrapper.
 // It has to be called before using the MultiRangeDownloader.
+// If lru cache is enabled and refCount was 0, the wrapper is removed from the cache.
 func (mrdWrapper *MultiRangeDownloaderWrapper) IncrementRefCount() {
 	mrdWrapper.mu.Lock()
 	defer mrdWrapper.mu.Unlock()
 
+	wasZero := mrdWrapper.refCount == 0
 	mrdWrapper.refCount++
 	if mrdWrapper.cancelCleanup != nil {
 		mrdWrapper.cancelCleanup()
 		mrdWrapper.cancelCleanup = nil
 	}
+
+	// If refCount was 0, remove from cache (file is being reopened)
+	if wasZero && mrdWrapper.mrdCache != nil {
+		deletedEntry := mrdWrapper.mrdCache.Erase(wrapperKey(mrdWrapper))
```

Comment:
why do we need this extra condition ? why can't the cache handle the not exists scenario?